pkg/apis/core/v1/defaults.go

```diff
@@ -456,7 +458,7 @@ func defaultPodRequests(obj *v1.Pod) {
 	// PodLevelResources feature) and pod-level requests are not set, the pod-level requests
 	// default to the effective requests of all the containers for that resource.
 	for key, aggrCtrLim := range aggrCtrReqs {
-		if _, exists := podReqs[key]; !exists && resourcehelper.IsSupportedPodLevelResource(key) {
+		if _, exists := podReqs[key]; !exists && resourcehelper.IsSupportedPodLevelResource(key) && !corev1helper.IsHugePageResourceName(key) {
```

Comment:
Please add a comment before the if statement stating why for huge pages the requests are not set equal to aggregated container requests